### ISSUE TITLE
Can not achieve 52.1 on VOC2012

### ISSUE THREAD

**Issue by swzhang5** [NVlabs/wetectron#44]
Hi, I train mist+cdb on VOC trainset+valset, and eval on test set.  config is voc/V_16_voc0712.yaml, but the result I got is only 37.5. Is there any tricks or notes I need to focus on? Thanks.

**Comment by jason718** [comment]
your results seems to be very low. I guess there might be some settings are wrong. Can you share your command, envs, and log?

**Comment by swzhang5** [comment]
Here is my log. I use 'det_old' to generate .txt files which are required by VOC website. But the result is very low. I also test on VOC2007 testset, the result is good(56.1) . 
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce RTX 3090
GPU 1: GeForce RTX 3090
GPU 2: GeForce RTX 3090
GPU 3: GeForce RTX 3090
GPU 4: GeForce RTX 3090
GPU 5: GeForce RTX 3090
GPU 6: GeForce RTX 3090
GPU 7: GeForce RTX 3090

Nvidia driver version: 455.45.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.7.1
[pip3] torchaudio==0.7.0a0+a853dff
[pip3] torchvision==0.8.2
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.0.221             h6bb024c_0  
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py37he8ac12f_0  
[conda] mkl_fft                   1.2.0            py37h23d657b_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] numpy                     1.19.2           py37h54aff64_0  
[conda] numpy-base                1.19.2           py37hfa32c7d_0  
[conda] pytorch                   1.7.1           py3.7_cuda11.0.221_cudnn8.0.5_0    pytorch
[conda] torchaudio                0.7.2                      py37    pytorch
[conda] torchvision               0.8.2                py37_cu110    pytorch
        Pillow (8.0.1)
2021-03-19 17:58:57,097 wetectron INFO: Loaded configuration file configs/voc/V_16_voc12.yaml
2021-03-19 17:58:57,098 wetectron INFO: 
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "catalog://VGGImageNetPretrained/JCJOHNS/VGG-16" 
  BACKBONE:
    CONV_BODY: "VGG16-OICR"
  WSOD_ON: True
  FASTER_RCNN: False
  ROI_HEADS:
    SCORE_THRESH: 0.0
    NMS: 0.3
  ROI_BOX_HEAD:
    NUM_CLASSES: 21
    POOLER_METHOD: "ROIPool"
    POOLER_RESOLUTION: 7
    POOLER_SCALES: (0.125,)
    FEATURE_EXTRACTOR: "VGG16.roi_head"
  ROI_WEAK_HEAD:
    PREDICTOR: "MISTPredictor"
    LOSS: "RoIRegLoss"
    OICR_P: 0.15
    REGRESS_ON: True
DB:
  METHOD: 'concrete'
DATASETS:
  TRAIN: ("voc_2012_train", "voc_2012_val")
  TEST: ("voc_2012_test",)
PROPOSAL_FILES:
  TRAIN: ('proposal/SS-voc_2012_train-boxes.pkl','proposal/SS-voc_2012_val-boxes.pkl')
  TEST:  ('proposal/SS-voc_2012_test-boxes.pkl',)
DATALOADER:
  SIZE_DIVISIBILITY: 32
INPUT:
  MIN_SIZE_TRAIN: (480, 576, 688, 864, 1000, 1200) 
  MAX_SIZE_TRAIN: 2000
  MIN_SIZE_TEST: 800
  MAX_SIZE_TEST: 2000
SOLVER:
  IMS_PER_BATCH: 16
  BASE_LR: 0.02
  WEIGHT_DECAY: 0.0001
  WARMUP_ITERS: 200
  STEPS: (20000, 26700)
  MAX_ITER: 30000
  CHECKPOINT_PERIOD: 1000
SOLVER_CDB:
  BASE_LR: 0.0001
  GAMMA: 3.0
  WEIGHT_DECAY: 0.0001
  STEPS: (20000, 26700)
TEST:
  BBOX_AUG:
    ENABLED: True
    HEUR: "AVG"
    H_FLIP: True
    SCALES: (480, 576, 688, 864, 1000, 1200) 
    MAX_SIZE: 2000
    SCALE_H_FLIP: True
SEED: 1234
2021-03-19 17:58:57,099 wetectron INFO: Running with config:
AMP_VERBOSE: False
DATALOADER:
  ASPECT_RATIO_GROUPING: True
  NUM_WORKERS: 4
  SIZE_DIVISIBILITY: 32
DATASETS:
  TEST: ('voc_2012_test',)
  TRAIN: ('voc_2012_train', 'voc_2012_val')
DB:
  GSM_THRES: 0.01
  IS_HARD: True
  LR: 0.01
  METHOD: concrete
  OPTIMIZER: sgd
  SIZE: 3
  TAU: 0.3
  WEIGHT: 0.01
DTYPE: float32
INPUT:
  BRIGHTNESS: 0.0
  CONTRAST: 0.0
  HUE: 0.0
  MAX_SIZE_TEST: 2000
  MAX_SIZE_TRAIN: 2000
  MIN_SIZE_TEST: 800
  MIN_SIZE_TRAIN: (480, 576, 688, 864, 1000, 1200)
  PCA: True
  PIXEL_MEAN: [102.9801, 115.9465, 122.7717]
  PIXEL_STD: [1.0, 1.0, 1.0]
  SATURATION: 0.0
  TO_BGR255: True
  VERTICAL_FLIP_PROB_TRAIN: 0.0
MODEL:
  BACKBONE:
    CONV_BODY: VGG16-OICR
    FREEZE_CONV_BODY_AT: 2
  CLS_AGNOSTIC_BBOX_REG: False
  DEVICE: cuda
  FASTER_RCNN: False
  FBNET:
    ARCH: default
    ARCH_DEF: 
    BN_TYPE: bn
    DET_HEAD_BLOCKS: []
    DET_HEAD_LAST_SCALE: 1.0
    DET_HEAD_STRIDE: 0
    DW_CONV_SKIP_BN: True
    DW_CONV_SKIP_RELU: True
    KPTS_HEAD_BLOCKS: []
    KPTS_HEAD_LAST_SCALE: 0.0
    KPTS_HEAD_STRIDE: 0
    MASK_HEAD_BLOCKS: []
    MASK_HEAD_LAST_SCALE: 0.0
    MASK_HEAD_STRIDE: 0
    RPN_BN_TYPE: 
    RPN_HEAD_BLOCKS: 0
    SCALE_FACTOR: 1.0
    WIDTH_DIVISOR: 1
  FPN:
    USE_GN: False
    USE_RELU: False
  GROUP_NORM:
    DIM_PER_GP: -1
    EPSILON: 1e-05
    NUM_GROUPS: 32
  KEYPOINT_ON: False
  MASK_ON: False
  META_ARCHITECTURE: GeneralizedRCNN
  RESNETS:
    BACKBONE_OUT_CHANNELS: 1024
    DEFORMABLE_GROUPS: 1
    NUM_GROUPS: 1
    RES2_OUT_CHANNELS: 256
    RES5_DILATION: 1
    STAGE_WITH_DCN: (False, False, False, False)
    STEM_FUNC: StemWithFixedBatchNorm
    STEM_OUT_CHANNELS: 64
    STRIDE_IN_1X1: True
    TRANS_FUNC: BottleneckWithFixedBatchNorm
    WIDTH_PER_GROUP: 64
    WITH_MODULATED_DCN: False
  RETINANET:
    ANCHOR_SIZES: (32, 64, 128, 256, 512)
    ANCHOR_STRIDES: (8, 16, 32, 64, 128)
    ASPECT_RATIOS: (0.5, 1.0, 2.0)
    BBOX_REG_BETA: 0.11
    BBOX_REG_WEIGHT: 4.0
    BG_IOU_THRESHOLD: 0.4
    FG_IOU_THRESHOLD: 0.5
    INFERENCE_TH: 0.05
    LOSS_ALPHA: 0.25
    LOSS_GAMMA: 2.0
    NMS_TH: 0.4
    NUM_CLASSES: 81
    NUM_CONVS: 4
    OCTAVE: 2.0
    PRE_NMS_TOP_N: 1000
    PRIOR_PROB: 0.01
    SCALES_PER_OCTAVE: 3
    STRADDLE_THRESH: 0
    USE_C5: True
  RETINANET_ON: False
  ROI_BOX_HEAD:
    CONV_HEAD_DIM: 256
    DILATION: 1
    FEATURE_EXTRACTOR: VGG16.roi_head
    MLP_HEAD_DIM: 1024
    NUM_CLASSES: 21
    NUM_STACKED_CONVS: 4
    POOLER_METHOD: ROIPool
    POOLER_RESOLUTION: 7
    POOLER_SAMPLING_RATIO: 0
    POOLER_SCALES: (0.125,)
    PREDICTOR: FastRCNNPredictor
    USE_GN: False
  ROI_HEADS:
    BATCH_SIZE_PER_IMAGE: 512
    BBOX_REG_WEIGHTS: (10.0, 10.0, 5.0, 5.0)
    BG_IOU_THRESHOLD: 0.5
    DETECTIONS_PER_IMG: 100
    FG_IOU_THRESHOLD: 0.5
    NMS: 0.3
    POSITIVE_FRACTION: 0.25
    SCORE_THRESH: 0.0
    USE_FPN: False
  ROI_KEYPOINT_HEAD:
    CONV_LAYERS: (512, 512, 512, 512, 512, 512, 512, 512)
    FEATURE_EXTRACTOR: KeypointRCNNFeatureExtractor
    MLP_HEAD_DIM: 1024
    NUM_CLASSES: 17
    POOLER_RESOLUTION: 14
    POOLER_SAMPLING_RATIO: 0
    POOLER_SCALES: (0.0625,)
    PREDICTOR: KeypointRCNNPredictor
    RESOLUTION: 14
    SHARE_BOX_FEATURE_EXTRACTOR: True
  ROI_MASK_HEAD:
    CONV_LAYERS: (256, 256, 256, 256)
    DILATION: 1
    FEATURE_EXTRACTOR: ResNet50Conv5ROIFeatureExtractor
    MLP_HEAD_DIM: 1024
    POOLER_RESOLUTION: 14
    POOLER_SAMPLING_RATIO: 0
    POOLER_SCALES: (0.0625,)
    POSTPROCESS_MASKS: False
    POSTPROCESS_MASKS_THRESHOLD: 0.5
    PREDICTOR: MaskRCNNC4Predictor
    RESOLUTION: 14
    SHARE_BOX_FEATURE_EXTRACTOR: True
    USE_GN: False
  ROI_WEAK_HEAD:
    CENTER_LOSS_ALPHA: 0.5
    CENTER_LOSS_WEIGHT: 0.003
    LOSS: RoIRegLoss
    OICR_P: 0.15
    PARTIAL_LABELS: none
    PREDICTOR: MISTPredictor
    REGRESS_HEUR: AVG
    REGRESS_ON: True
    ROI_LOSS_REFINE: False
    USE_TEMPLATE: False
  RPN:
    ANCHOR_SIZES: (32, 64, 128, 256, 512)
    ANCHOR_STRIDE: (16,)
    ASPECT_RATIOS: (0.5, 1.0, 2.0)
    BATCH_SIZE_PER_IMAGE: 256
    BG_IOU_THRESHOLD: 0.3
    FG_IOU_THRESHOLD: 0.7
    FPN_POST_NMS_PER_BATCH: True
    FPN_POST_NMS_TOP_N_TEST: 2000
    FPN_POST_NMS_TOP_N_TRAIN: 2000
    MIN_SIZE: 0
    NMS_THRESH: 0.7
    POSITIVE_FRACTION: 0.5
    POST_NMS_TOP_N_TEST: 1000
    POST_NMS_TOP_N_TRAIN: 2000
    PRE_NMS_TOP_N_TEST: 6000
    PRE_NMS_TOP_N_TRAIN: 12000
    RPN_HEAD: SingleConvRPNHead
    STRADDLE_THRESH: 0
    USE_FPN: False
  RPN_ONLY: False
  WEIGHT: catalog://VGGImageNetPretrained/JCJOHNS/VGG-16
  WSOD_ON: True
OUTPUT_DIR: /home/ZhangShiwei/wetectron/12
PATHS_CATALOG: /home/ZhangShiwei/wetectron/wetectron/config/paths_catalog.py
PROPOSAL_FILES:
  TEST: ('proposal/SS-voc_2012_test-boxes.pkl',)
  TRAIN: ('proposal/SS-voc_2012_train-boxes.pkl', 'proposal/SS-voc_2012_val-boxes.pkl')
SEED: 1234
SOLVER:
  BASE_LR: 0.02
  BIAS_LR_FACTOR: 2
  CHECKPOINT_PERIOD: 1000
  GAMMA: 0.1
  IMS_PER_BATCH: 16
  ITER_SIZE: -1
  MAX_ITER: 30000
  MOMENTUM: 0.9
  STEPS: (20000, 26700)
  WARMUP_FACTOR: 0.3333333333333333
  WARMUP_ITERS: 200
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0001
  WEIGHT_DECAY_BIAS: 0
SOLVER_CDB:
  BASE_LR: 0.0001
  BIAS_LR_FACTOR: 2
  GAMMA: 3.0
  MOMENTUM: 0.9
  STEPS: (20000, 26700)
  WARMUP_FACTOR: 0.3333333333333333
  WARMUP_ITERS: 500
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0001
  WEIGHT_DECAY_BIAS: 0
TEST:
  BBOX_AUG:
    ENABLED: True
    HEUR: AVG
    H_FLIP: True
    MAX_SIZE: 2000
    SCALES: (480, 576, 688, 864, 1000, 1200)
    SCALE_H_FLIP: True
  DETECTIONS_PER_IMG: 100
  EXPECTED_RESULTS: []
  EXPECTED_RESULTS_SIGMA_TOL: 4
  IMS_PER_BATCH: 8
  VIS_THRES: 0.5

**Comment by jason718** [comment]
this seems to be some format issues. You might want to check their formats or your results again, eg., visualize some predictions. 

Our voc 2012 testing was fine, http://host.robots.ox.ac.uk:8080/anonymous/DCJ5GA.html. I cannot recall any strange things.

voc 2012 also has validation set. you can always try stuff on that split without too much pressure.

feel free to reopen.
